PxtlCa.XmlCommentMarkDownGenerator.MSBuild/GenerateMarkdown.cs

```diff
@@ -0,0 +1,273 @@
+using System.Text;
+//using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.IO;
+using System.Xml.Linq;
+using System.Xml;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+using PxtlCa.XmlCommentMarkDownGenerator.MSBuild.Options;
+
+namespace PxtlCa.XmlCommentMarkDownGenerator.MSBuild
+{
+    /// <summary>
+    /// A task that generates and optionally merges markdown
+    /// </summary>
+    public class GenerateMarkdown : Task
+    {
+        /// <summary>
+        /// The file(s) from which to generate markdown.  This should be in XmlDocumentation format.
+        /// </summary>
+        [Required]
+        public ITaskItem[] InputXml { get; set; }
+
+        /// <summary>
+        /// DocumentationPath is the top level directory in which to search for files.
+        /// It is also the path where generated markdown files are created.
+        /// </summary>
+        [Required]
+        public ITaskItem DocumentationPath { get; set; }
+
+        /// <summary>
+        /// Whether the generated markdown files should merge.  Only valid if multiple markdown files exist.
+        /// DocumentationPath is the top level directory in which to search for files.
+        /// Both existing markdown files and the generated files are merged.
+        /// </summary>
+        public bool MergeFiles { get; set; }
+
+        /// <summary>
+        /// The file to be created by the merge.  Unused if MergeFiles evaluates to false.
+        /// </summary>
+        public ITaskItem OutputFile { get; set; }
+
+        /// <summary>
+        /// Defaults to false. When true unexpected tags in the documentation
+        /// will generate warnings rather than errors. 
+        /// </summary>
+        public bool WarnOnUnexpectedTag { get; set; } = false;
+
+      
+
+        /// <summary>
+        /// Runs the task as configured
+        /// </summary>
+        /// <returns>true if task has succeeded</returns>
+        public override bool Execute()
+        {
+            if (InputXml.Length == 0)
+            {
+                Log.LogError("InputXml cannot be empty");
+            }
+            else
+            {
+                UpdateParametersFromInput();
+                if (MergeFiles && (null == OutputFile))
+                {
+                    Log.LogError("OutputFile must be specified if input files are merged");
+                }
+                else if (DocumentationPathIsFile && InputXml.Length != 1)
+                {
+                    Log.LogError("DocumentationPath must specify a directory if more than one input XML value is supplied");
+                }
+                else
+                {
+                    try
+                    {
+                        CreateDirectoryIfNeeded();
+                        GenerateFiles();
+                        if (MergeFiles)
+                        {
+                            Merge();
```

Comment:
Shouldn't there be a separate CreateDirectoryIfNeeded for outputfile because it's possible that the merged outputfile is *outside* of documentation path?